File: python/jzoffer-62.py
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    # 返回对应节点TreeNode
    def KthNode(self, pRoot, k):
        # write code here
        self.res = []
        self.mid(pRoot)
        return self.res[k-1] if 0<k<=len(self.res) else None
    
    def mid(self, root):
        if not root:
            return
        self.mid(root.left)
        self.res.append(root)
        self.mid(root.right)

File: java/jzoffer-62.java
/*
public class TreeNode {
    int val = 0;
    TreeNode left = null;
    TreeNode right = null;

    public TreeNode(int val) {
        this.val = val;

    }

}
*/
public class Solution {
    int index = 0;
    TreeNode KthNode(TreeNode root, int k)
    {
        if(root != null){
            TreeNode node = KthNode(root.left,k);
            if(node != null)
                return node;
            index ++;
            if(index == k)
                return root;
            node = KthNode(root.right,k);
            if(node != null)
                return node;
        }
        return null;
    }
}